src/arrays.js

```diff
@@ -3,38 +3,80 @@
 
 
 const each = (elements, cb) => {
+  for (let i = 0; i < elements.length; i++) {
+    cb(elements[i], i);
+  }
   // Iterates over a list of elements, yielding each in turn to the `cb` function.
   // This only needs to work with arrays.
   // based off http://underscorejs.org/#each
 };
 
 const map = (elements, cb) => {
+  const array = [];
+  for (let i = 0; i < elements.length; i++) {
+    array.push(cb(elements[i]));
+  }
+  return array;
   // Produces a new array of values by mapping each value in list through a transformation function (iteratee).
   // Return the new array.
 };
 
 const reduce = (elements, cb, memo) => {
+  let i = 0;
+  if (!memo) {
+    memo = elements[0];
+    i = 1;
+  }
+  let total = memo; // 10
+  for (; i < elements.length; i++) {
+    total += elements[i];
```

Comment:
You wouldn't want to use `+=` here.  That is doing addition or string concatenation.  You would want to actually overwrite the `memo`.  `memo = cb(memo, elements[i]);`